jscomp/others/node_buffer.ml

```diff
@@ -26,17 +26,208 @@
+  unit -> t = "fill"
+[@@bs.send]
+
+external includes :
+  t -> 
+  ?byteOffset:int ->
+  ?encoding: ([ `ascii  | `utf8  | `utf16le  | `usc2  | `base64  | `latin1 | `binary | `hex ] [@bs.string]) ->
+  unit -> bool = "includes"
+[@@bs.send]
+
+external indexOf :
+  t -> 
+  ([ `String of string | `Integer of int | `Buffer of t ] [@bs.unwrap]) ->
+  ?byteOffset:int ->
+  ?encoding: ([ `ascii  | `utf8  | `utf16le  | `usc2  | `base64  | `latin1 | `binary | `hex ] [@bs.string]) ->
+  unit -> int = "indexOf"
+[@@bs.send]
+
+external lastIndexOf :
+  t -> 
+  ([ `String of string | `Integer of int | `Buffer of t ] [@bs.unwrap]) ->
+  ?byteOffset:int ->
+  ?encoding: ([ `ascii  | `utf8  | `utf16le  | `usc2  | `base64  | `latin1 | `binary | `hex ] [@bs.string]) ->
+  unit -> int = "lastIndexOf"
+[@@bs.send]
+
+external length : t -> int = "length" 
+[@@bs.get]
+
+external readDoubleBE : t -> int -> float = "readDoubleBE" [@@bs.send]
+external readDoubleLE : t -> int -> float = "readDoubleLE" [@@bs.send]
+external readFloatBE : t -> int -> float = "readFloatBE" [@@bs.send]
+external readFloatLE : t -> int -> float = "readFloatLE" [@@bs.send]
+
+external readInt8 : t -> int -> int = "readInt8" [@@bs.send]
+external readInt16BE : t -> int -> int = "readInt16BE" [@@bs.send]
+external readInt16LE : t -> int -> int = "readInt16LE" [@@bs.send]
+external readInt32BE : t -> int -> int = "readInt32BE" [@@bs.send]
+external readInt32LE : t -> int -> int = "readInt32LE" [@@bs.send]
+external readIntBE : t -> offset:int -> byteLength:int -> int = "readIntBE" [@@bs.send]
+external readIntLE : t -> offset:int -> byteLength:int -> int = "readIntLE" [@@bs.send]
+
+external readUInt8 : t -> int -> int = "readUInt8" [@@bs.send]
+external readUInt16BE : t -> int -> int = "readUInt16BE" [@@bs.send]
+external readUInt16LE : t -> int -> int = "readUInt16LE" [@@bs.send]
+external readUInt32BE : t -> int -> int = "readUInt32BE" [@@bs.send]
+external readUInt32LE : t -> int -> int = "readUInt32LE" [@@bs.send]
+external readUIntBE : t -> offset:int -> byteLength:int -> int = "readUIntBE" [@@bs.send]
+external readUIntLE : t -> offset:int -> byteLength:int -> int = "readUIntLE" [@@bs.send]
+
+external subarray : t -> ?start: int -> ?end_: int -> unit -> t = "subarray" [@@bs.send]
+external slice : t -> ?start: int -> ?end_:int -> unit -> t = "slice" [@@bs.send]
+
+external swap16 : t -> t = "swap16" [@@bs.send]
+external swap32 : t -> t = "swap32" [@@bs.send] 
+external swap64 : t -> t = "swap64" [@@bs.send] 
+
+external toJSON : t -> Js.Json.t = "toJSON" [@@bs.send]
+
+external write :
+  t ->
+  string ->
+  ?offset:int ->
+  ?length:int ->
+  ?encoding: ([ `ascii  | `utf8  | `utf16le  | `usc2  | `base64  | `latin1 | `binary | `hex ] [@bs.string]) ->
+  unit -> int = "write" [@@bs.send]
+
+external writeDoubleBE : t -> float -> offset:int -> int = "writeDoubleBE" [@@bs.send]
```

Comment:
Looks like `offset` should be an optional param in most of these cases. The table of contents on the doc page will actually give you a quick overview of them all. Basically wherever it shows `[, offset]` you can use `?offset:int`.

Reply:
Initially, I've made all of them optional, but at one moment I started to think that 90% of the time when you read and write to buffer, you specify position, since it defaults to 0 when unspecified, so in the end I've got it as required arg. Will change back to follow JS side more strictly, thanks.

Reply:
Made them optional again.